Add automated tests for Gitea, Jenkins, and Nostr applications

khaledyoussef24 opened the pull request:
### Description

Added automated tests for gitea to check domain is reachable
Added a test for nostr to check if it prints at port 8080 the "use nostr client message"
Added a test for jenkins to check if the domain is reachable
https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3702

### Changes

List of changes this PR includes

### Related Issues

List of related issues

### Tested Scenarios

A list of scenarios tried to match the deliverables

### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### To consider

#### Preliminary Checks:

- Does it completely address the issue linked?
- What about edge cases?
- Does it meet the specified acceptance criteria?
- Are there any unintended side effects?
- Does the PR adhere to the team's coding conventions, style guides, and best practices?
- Does it integrate well with existing features?
- Does it impact the overall performance of the application?
- Are there any bottlenecks or slowdowns?
- Has it been optimized for efficiency?
- Has it been adequately tested with unit, integration, and end-to-end tests?
- Are there any known defects or issues?
- Is the code well-documented?
- Are changes to documentation reflected in the code?

#### UI Checks:

- If a UI design is provided/ does it follow it?
- Does every button work?
- Is the data displayed logical? Is it what you expected?
- Does every validation work?
- Does this interface feel intuitive?
- What about slow network? Offline?
- What if the gridproxy/graphql/chain is failing?
- What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- Code formatted/linted? Are there unused imports? .. etc
- Is there redundant/repeated code?
- Are there conditionals that are always true or always false?
- Can we write this more concisely?
- Can we reuse this code? If yes, where?
- Will the changes be easy to maintain and update in the future?
- Can this code become too complex to understand for other devs?
- Can this code cause future integration problems?

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

khaledyoussef24 committed:
Add automated tests for Gitea, Jenkins, and Nostr applications

khaledyoussef24 committed:
adding ufw for the working port

khaledyoussef24 committed:
Update deployment scripts for Gitea, Jenkins, and Nostr

khaledyoussef24 committed:
Added gitea test, removed outdated gittea test, and modified jenkins and nostr tests

khaledyoussef24 commented:
Work done : 
add test details and testcases to the 3 tests
stuck on the nostor test still re-checking wether to `curl localhost:port` and to verify the appering message is enough or not

https://app.testlodge.com/a/26076/projects/40893/suites/194096?expand_section=366955#case_4427212
https://app.testlodge.com/a/26076/projects/40893/suites/194096?expand_section=366955#case_4427214
https://app.testlodge.com/a/26076/projects/40893/suites/194096?expand_section=366955#case_4427218

khaledyoussef24 committed:
fixing tests and making relative path for the ssh key

0oM4R reviewed (changes requested):
the three tests is failing 
<img width="1183" alt="image" src="https://github.com/user-attachments/assets/17824d29-1493-4827-a045-7d85c7f4d832" />
<img width="1208" alt="image" src="https://github.com/user-attachments/assets/32203073-58fa-4135-8a53-0de9a035b17d" />


this ran on the workflow, https://github.com/threefoldtech/tfgrid-sdk-ts/runs/35041354783

khaledyoussef24 committed:
fixes for ssh key

khaledyoussef24 commented:
fixed the issues shall work now fine @0oM4R

0oM4R commented:
tests failed due to lerna version; please merge development branch

maayarosama commented on packages/grid_client/tests/modules/applications/gitea.test.ts:
```diff
@@ -0,0 +1,212 @@
+
+  // Gateway Node Selection
+  const gatewayNodes = await gridClient.capacity.filterNodes({
+    gateway: true,
```
add features in the filters to avoid retrieving nodes that don't align with required features

maayarosama commented on packages/grid_client/tests/modules/applications/gitea.test.ts:
```diff
@@ -0,0 +1,212 @@
+});
+
+// Private IP Regex
+const ipRegex = /(^127\.)|(^10\.)|(^172\.1[6-9]\.)|(^172\.2[0-9]\.)|(^172\.3[0-1]\.)|(^192\.168\.)/;
```
Whys do we need this variable? It's not used in any tests

maayarosama commented on packages/grid_client/tests/modules/applications/gitea.test.ts:
```diff
@@ -0,0 +1,212 @@
+
+  // Node Selection
+  const nodes = await gridClient.capacity.filterNodes({
+    cru: cpu,
```
Add features

maayarosama commented on packages/grid_client/tests/modules/applications/nostr.test.ts:
```diff
@@ -0,0 +1,191 @@
+  if (nodeId == -1) throw new Error("No nodes available to complete this test");
+  const domain = name + "." + GatewayNode.publicConfig.domain;
+
+  const fs = require("fs");
```
Why do we need this?

maayarosama commented on packages/grid_client/tests/modules/applications/gitea.test.ts:
```diff
@@ -0,0 +1,212 @@
+  if (nodeId === -1) throw new Error("No nodes available to complete this test");
+  const domain = name + "." + GatewayNode.publicConfig.domain;
+
+  const fs = require("fs");
```
why do we need this?

maayarosama commented on packages/grid_client/tests/modules/applications/jenkins.test.ts:
Add Features to filters, ipRegex and fs aren't needed

maayarosama commented on packages/grid_client/tests/modules/applications/nostr.test.ts:
Add Features to filters, ipRegex and fs aren't needed

maayarosama commented:
Nostr and gitea tests are failing 
![Screenshot from 2025-01-16 14-10-37](https://github.com/user-attachments/assets/cd544170-4e2e-4a5a-817d-9f5bb9df6880)
![Screenshot from 2025-01-16 14-14-28](https://github.com/user-attachments/assets/3bc4e967-63bb-4030-bfa1-9dc5c03c4d56)

khaledyoussef24 committed:
Merge branch 'development' into development_adding_grid_client_tests

khaledyoussef24 committed:
fixing tests and adding features to node finder

khaledyoussef24 committed:
fixes for tests

maayarosama commented on packages/grid_client/tests/modules/applications/nostr.test.ts:
tests are failing 
![Screenshot from 2025-01-26 11-05-01](https://github.com/user-attachments/assets/6c92f3db-ecca-476a-8f66-0ebbd450cb75)

0oM4R replied:
https://github.com/threefoldtech/tfgrid-sdk-ts/runs/36340717094

khaledyoussef24 committed:
Merge branch 'development' of https://github.com/threefoldtech/tfgrid-sdk-ts into development_adding_grid_client_tests

khaledyoussef24 committed:
fixing jenkins

khaledyoussef24 committed:
fixing jenkins

khaledyoussef24 commented:
fixed jenkins 
debugged to find out that i was using a false port at the first but even after this the test was failing but found out the machine were working fine with the flist but the issue that the script was pinging the url and waiting for code success from it
<img width="1653" alt="Screenshot 2025-02-20 at 9 00 18 AM" src="https://github.com/user-attachments/assets/72aaafaa-f5f2-4294-9620-451a93e44a09" />

while the url  redirects to login page "/login?from=%2F"
<img width="1680" alt="Screenshot 2025-02-20 at 9 02 16 AM" src="https://github.com/user-attachments/assets/82df6912-34da-4127-b97c-885d830c1e5b" />
added it to the script and it finally worked fine
<img width="1120" alt="Screenshot 2025-02-20 at 8 59 34 AM" src="https://github.com/user-attachments/assets/87aa4ff7-0b7c-450c-8daa-39af3a0483e6" />

khaledyoussef24 committed:
fixed port of the machine for the jenkins script and the link of the gateway redirection

khaledyoussef24 committed:
fixes for the nostr test

khaledyoussef24 commented:
work done fixed both nostr and jenkins 
and are working fine now
the issue in nostr test was in network selections
fixed it and added wire guard access
<img width="371" alt="Screenshot 2025-02-20 at 3 49 09 PM" src="https://github.com/user-attachments/assets/2ad4b20a-2779-4832-8a3c-3253415d3d25" />
<img width="681" alt="Screenshot 2025-02-20 at 3 49 27 PM" src="https://github.com/user-attachments/assets/21162fd4-4cd4-46da-8c77-70ce6a8b4fcf" />

0oM4R commented on packages/grid_client/tests/modules/applications/gitea.test.ts:
```diff
@@ -0,0 +1,203 @@
+        flist: "https://hub.grid.tf/petep.3bot/threefolddev-gitea-latest.flist",
+        entrypoint: "/sbin/zinit init",
+        public_ip: publicIp,
+        planetary: true,
```
planetary is not added to feature; make sure to add it you may find it as yggdrasil

0oM4R commented on packages/grid_client/tests/modules/applications/jenkins.test.ts:
```diff
@@ -0,0 +1,206 @@
+        flist: "https://hub.grid.tf/tf-official-apps/jenkins-latest.flist",
+        entrypoint: "/sbin/zinit init",
+        public_ip: publicIp,
+        planetary: true,
```
also here ygg is not included in the features

0oM4R commented on packages/grid_client/tests/modules/applications/jenkins.test.ts:
```diff
@@ -0,0 +1,206 @@
+  const vms: MachinesModel = {
+    name: deploymentName,
+    network: {
+      name: networkName,
```
should we add access ? as we are using wireguard?

khaledyoussef24 replied:
removed wireguard so it is no longer needed

0oM4R reviewed (commented):
are you sure that the machines have a wireguard?
i suggest to add test case for that, or remove it from features

khaledyoussef24 committed:
fixing features

0oM4R reviewed (approved):
tests pass on workflow https://github.com/threefoldtech/tfgrid-sdk-ts/runs/39835841421
